tfe/tfe_sweeper_test.go

```diff
@@ -0,0 +1,59 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/go-tfe"
+	"log"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+func TestMain(m *testing.M) {
+	resource.TestMain(m)
+}
+
+func getOrgSweeper(name string) *resource.Sweeper {
+	return &resource.Sweeper{
+		Name: name,
+		F: func(ununsed string) error {
+			if os.Getenv("TFE_HOSTNAME") == "" && (os.Getenv("TFE_TOKEN") == "") {
+				return fmt.Errorf("must provide environment variables TFE_HOSTNAME and TFE_TOKEN")
+			}
+			hostname := os.Getenv("TFE_HOSTNAME")
+			token := os.Getenv("TFE_TOKEN")
+			insecure := os.Getenv("TFE_INSECURE") != ""
+
+			client, err := getClient(hostname, token, insecure)
+			if err != nil {
+				return err
+			}
+
+			ctx := context.TODO()
+			orgList, err := client.Organizations.List(ctx, tfe.OrganizationListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, org := range orgList.Items {
+				log.Printf("[DEBUG] Testing if org %s starts with tst-terraform or named terraform-updated", org.Name)
+				if strings.HasPrefix(org.Name, "tst-terraform") || org.Name == "terraform-updated" {
+					log.Printf("[DEBUG] deleting org %s", org.Name)
+					err = client.Organizations.Delete(ctx, org.Name)
+					if err != nil {
+						return err
+					}
+				}
+			}
+			return nil
+		},
+	}
+}
+
+// Sweepers usually go along with the tests. In TF[CE]'s case everything depends on the organization,
+// which means that if we delete it then all the other entities will  be deleted automatically.
+func init() {
+	resource.AddTestSweepers("org_sweeper", getOrgSweeper("org_sweeper"))
+}
```

Comment:
📚 just for my own learning, since i don't know very much about how sweepers work:
does putting the sweeper here instead of in the organization acceptance test mean that this sweeper will run after every set of acceptance tests?

Reply:
The sweepers run only when you run `make sweep`. Usually you'd add a `sweepers` directory and create a separate file for each sweeper. In our case the way TF[CE] models things we only need to delete the org and the rest will follow automatically, so it made sense to me to put it in this file.